hazelcast/src/main/java/com/hazelcast/internal/management/ManagementCenterService.java

```diff
@@ -294,61 +284,30 @@ private boolean post(HttpURLConnection connection) throws IOException {
         return responseCode == HTTP_SUCCESS;
     }
 
-    private final class PrepareStateThread extends Thread {
```

Comment:
Don't see the added value to have an additional thread in place. 
The old approach is fragile because it relies on these 2 threads to be executed in a particular order.  Due to scheduling irregularities, the old approach can lead to the SendThread sending the old state twice if the PrepareStateThread has not yet executed.